app/react/Metadata/components/GroupedGeolocationViewer.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import GeolocationViewer from 'app/Metadata/components/GeolocationViewer';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { Translate } from 'app/I18N';
+import { IStore } from '../../istore';
+import { ensure } from '../../../shared/tsUtils';
+import { Pill } from './Pill';
+
+interface GeolocationMarker {
+  lat: string;
+  lon: string;
+  label?: string;
+  color?: string;
+}
+
+interface GroupMember {
+  label: string;
+  name: string;
+  translateContext: string;
+  value: GeolocationMarker[];
+}
+
+interface TemplatesInfo {
+  [template: string]: {
+    name: string;
+    color: string;
+  };
+}
+
+export interface GroupedGeolocationViewerProps {
+  members: GroupMember[];
+  templatesInfo: TemplatesInfo;
+}
+
+const notLabeledOrMultiple = (member: GroupMember) =>
+  member.value.length === 1 && !member.value[0].label;
+
+const getFirstGroupInfo = (members: GroupMember[], templatesInfo: TemplatesInfo) => (
+  <dl
+    className="pills-container"
+    key={members.map(member => `${member.translateContext}_${member.name}`).join(',')}
+  >
+    <dt />
+    <dd>
+      {members.map(member => (
+        <Pill
+          key={`${member.value[0].lat}_${member.value[0].lon}`}
+          color={templatesInfo[member.translateContext].color}
+        >
+          <Translate context={member.translateContext}>{member.label}</Translate>
+        </Pill>
+      ))}
+    </dd>
+  </dl>
+);
+
+const getMultiMemberInfo = (templatesInfo: TemplatesInfo) => (member: GroupMember) => (
+  <dl className="pills-container" key={`${member.translateContext}_${member.name}`}>
+    <dt>
+      <span>
+        <Translate context={member.translateContext}>{member.label}</Translate>
+        <> (</>
+        <Translate>linked</Translate>
+        <> </>
+        <Translate context={member.translateContext}>
+          {templatesInfo[member.translateContext].name}
+        </Translate>
+        <>)</>
+      </span>
+    </dt>
+    <dd>
+      {member.value.map(value => (
+        <Pill
+          key={`${value.lat}_${value.lon}`}
+          color={templatesInfo[member.translateContext].color}
+        >
+          <Translate context={member.translateContext}>{value.label || ''}</Translate>
+        </Pill>
+      ))}
+    </dd>
+  </dl>
+);
+
+const computeRenderMemberGroups = (members: GroupMember[], templatesInfo: TemplatesInfo) => {
+  const firstGroup: GroupMember[] = [];
+  const restOfGroups: GroupMember[] = [];
+
+  members.forEach(member => {
+    if (notLabeledOrMultiple(member)) {
+      firstGroup.push(member);
+    } else {
+      restOfGroups.push(member);
+    }
+  });
+
+  return [getFirstGroupInfo(firstGroup, templatesInfo)].concat(
+    restOfGroups.map(getMultiMemberInfo(templatesInfo))
+  );
+};
+
+// eslint-disable-next-line react/no-multi-comp
```

Comment:
I'm getting this eslint error because I splitted the "render function", so now I have more than one function returning JSX and eslint understand I'm defining more than one component in the same file. I technically AM but, do you think it's worth making a brand new component out of something so tightly coupled to the "main one"? Please, let me know your thoughts!!

Reply:
How about moving the three complementary functions within the main function component? This is valid and you wouldn't need to ignore the eslint rule.